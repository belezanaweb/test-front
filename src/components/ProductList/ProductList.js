import React, { useState, useEffect, useContext } from 'react'
import { CartContext } from '../../store/CartStore'
import { ProductListWrapper, ProductListSection, ProductListTitle } from './ProductList.styles'
import { ProductCard } from '../ProductCard/ProductCard'

export const ProductList = ({ title = 'Produtos' }) => {
  const cartContext = useContext(CartContext)
  const [products, setProducts] = useState([])

  useEffect(() => {
    /* istanbul ignore else */
    if (cartContext.state.items) {
      setProducts(
        cartContext.state.items.map(({ product }) => ({
          productTitle: product.name,
          productPrice: product.priceSpecification.price,
          imageUrl: product.imageObjects[0].small
        }))
      )
    }
  }, [cartContext])

  return (
    <ProductListSection>
      <ProductListTitle>{title}</ProductListTitle>
      <ProductListWrapper>
        {products.map((product) => (
          <ProductCard key={product.productTitle} {...product} />
        ))}
      </ProductListWrapper>
    </ProductListSection>
  )
}
