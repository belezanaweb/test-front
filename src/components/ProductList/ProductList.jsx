import React from 'react'

export default function ProductList() {
  return (
    <ul>
      <li>3</li>
      <li>2</li>
      <li>1</li>
    </ul>
  )
}
