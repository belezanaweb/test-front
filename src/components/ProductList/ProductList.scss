@import '../../assets/sass/colors';

.product-list {
  width: 100%;
  height: auto;
  display: flex;
  align-items: flex-start;
  justify-content: flex-start;
  margin: 0;
  position: relative;
  border-radius: 3px;
  border: 1px solid #EEEEEE;
  padding: 11px 10px 14px 10px;
  &__thumb {
    width: 65px;
    height: 65px;
    display: block;
    flex: none;
    img {
      width: 100%;
    }
  }
  &__content {
    width: 100%;
    height: auto;
    position: relative;
    margin: 0;
    padding-left: 11px;
  }
  &__title {
    font-size: 13px;
    line-height: 16px;
    font-weight: 400;
    color: $graphiteColor;
  }
  &__price {
    font-weight: 700;
    color: $graphiteColor;
    width: 100%;
    text-align: right;
    display: block;
    margin-top: 15px;
    font-size: 14px;
    line-height: 17px;
  }
}
