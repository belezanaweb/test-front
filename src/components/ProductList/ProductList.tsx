import styled, { css } from 'styled-components';

import { Image, ImageWrapper, ItemPrice } from './ProductListItem';

interface ProductListProps {
  size?: 'sm' | 'md';
}

const ProductList = styled.div<ProductListProps>`


  ${({ size }) => {
    switch (size) {
      case 'sm':
        return css`
          ${ImageWrapper} {
            flex: 0 42px;
          }

          ${Image} {
            height: 42px;
          }

          ${ItemPrice} {
            display: none;
          }
        `;
      case 'md':
      default:
        return css``;
    }
  }}
`;

ProductList.defaultProps = {
  size: 'md',
};

export default ProductList;