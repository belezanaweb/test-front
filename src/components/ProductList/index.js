import React from 'react';
import './styles.css';
import ItemCard from '../ItemCard';
import Loading from '../Loading';

const ProductList = props => {
    return (
        <>
            {props.store.items ? props.store.items.map((element, index) =>
                <ItemCard key={index} item={element} />
            )
                :
                <Loading />
            }
        </>
    )
}

export default ProductList;

