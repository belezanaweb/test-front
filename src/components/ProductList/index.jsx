import React from 'react'

const ProductList = () => {
  return <></>
}

export default ProductList
