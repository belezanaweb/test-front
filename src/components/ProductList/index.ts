export { default } from './ProductList';
export { default as ProductListItem } from './ProductListItem';
