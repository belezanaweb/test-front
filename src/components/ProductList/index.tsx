import React from 'react'
import { useSelector } from 'react-redux'
import { IOrder } from '../../model/order'
import { ProductItemStyled } from '../../pages/ProductBag/styles'
import { RootState } from '../../redux'
import { ContentStyled } from '../../styles/global'

const ProductList: React.FC = () => {
  const order = useSelector<RootState, IOrder | undefined>((state) => {
    return state.orderReducer.order
  })
  return (
    <>
      <h3>Produtos</h3>
      <ContentStyled>
        {order &&
          order.items.length > 0 &&
          order.items.map((item) => {
            return (
              <ProductItemStyled key={item.product.sku}>
                <img src={item.product.imageObjects[0].small} alt={item.product.name} />
                <div>{item.product.name.substring(0, 60)}</div>
              </ProductItemStyled>
            )
          })}
      </ContentStyled>
    </>
  )
}

export default ProductList
