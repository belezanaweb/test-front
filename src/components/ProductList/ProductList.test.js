import React from 'react'
import { render } from '@testing-library/react'
import { CartStore } from '../../store/CartStore'
import { ProductList } from './ProductList'

const CartStoreValue = {
  state: {
    items: [
      {
        product: {
          name: 'Teste',
          priceSpecification: {
            price: 123.45
          },
          imageObjects: [
            {
              small: 'http://localhost.test'
            }
          ]
        }
      }
    ]
  }
}

test('renders PaymentForm', () => {
  const { container } = render(
    <CartStore value={CartStoreValue}>
      <ProductList />
    </CartStore>
  )

  expect(container).toMatchSnapshot()
})
