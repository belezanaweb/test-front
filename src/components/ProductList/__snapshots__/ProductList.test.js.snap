// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders PaymentForm 1`] = `
.c1 {
  padding: 12px 12px 13px 13px;
  border-radius: 3px;
  background-color: #fff;
  box-shadow: 1px 1px 5px 0 rgba(0,0,29,0.22);
}

.c0 {
  display: block;
  margin: 0 0 5px 13px;
  color: #999;
  font-size: 14px;
  font-weight: 700;
  line-height: 1.21428571428571;
  text-transform: uppercase;
}

.c2 {
  display: grid;
  grid-gap: 15px;
}

.c3 {
  display: grid;
  grid-gap: 11px 15px;
  grid-template-columns: 65px 1fr;
  grid-template-rows: 1fr 1fr;
  padding: 12px 10px 14px;
  border: 1px solid #eee;
  border-radius: 3px;
}

.c4 {
  width: 65px;
  height: 65px;
  object-fit: contain;
  object-position: 50% 100%;
  grid-row: span 2;
}

.c5 {
  margin: 0;
  color: #212122;
  font-size: 13px;
  font-weight: normal;
  line-height: 1.23076923076923;
}

.c6 {
  display: block;
  font-size: 14px;
  font-weight: 700;
  line-height: 1.21428571428571;
  text-align: right;
}

<div>
  <section
    class=""
  >
    <h1
      class="c0"
    >
      Produtos
    </h1>
    <div
      class="c1 c2"
    >
      <article
        class="c3"
      >
        <img
          alt="Teste"
          class="c4"
          src="http://localhost.test"
        />
        <h2
          class="c5"
        >
          Teste
        </h2>
        <span
          class="c6"
        >
          R$Â 123.45
        </span>
      </article>
    </div>
  </section>
</div>
`;
